Revamped React 1

chezzoba opened the pull request:
The curriculum is now completely revamped and ready for testing on the guinea pigs we call team28!

For a summary of the revamped React curriculum and lesson plans view [this planning document](https://docs.google.com/document/d/1pEi5nyYAr_kesPMva_eF747rxaOX3J9IgeZ63nNIKNw).

adamblanchard commented on react1/week1/lesson-plan.md:
```diff
@@ -1,207 +1,164 @@
+### 5. Setting properties such as className and value (15 minutes)
+
+- Create a functional component called `Button` that renders a `button` element with a `className` prop set to `btn btn-primary`.
+- Create a functional component called `DangerButton` that renders a `button` element with a dangerous text and a `className` prop set to `btn btn btn-primary`.
```
Should this just be 'btn btn-primary'?

adamblanchard commented on react1/week1/homework.md:
```diff
@@ -1,34 +1,2 @@
-
-Need to brush up on the homework hand-in process?<br/>
-Check [this resource](https://github.com/HackYourFuture-CPH/Git/blob/main/homework_hand_in.md) to remember how to hand in the homework correctly!
+Solve the week one tasks in the [homework project](/react1/homework/app/about_us/README.md)
```
This link is broken

adamblanchard commented on readme.md:
```diff
@@ -1,69 +1,59 @@
 
-- **[FlowType](https://flowtype.org)** - a way to annotate ES6 with types, similar to TypeScript
+- **[TypeScript](https://www.typescriptlang.org/)** - a way to annotate ES6 with types
 - **[Redux](https://flowtype.org)** - predictable (application) state container, a Flux implementation
```
Could you fix the Redux link while you're here? :D

adamblanchard commented on react1/week2/preparation.md:
```diff
@@ -1,10 +1,4 @@
-- [React 1 week 2 part 1](https://youtu.be/ArGRhOgf5LU)
-- [React 1 week 2 part 2](https://youtu.be/E_5U6lzI8-A)
+- [Using the State Hook](https://react.dev/reference/react/useState) (20 min)
+- https://vercel.com/blog/nextjs-server-side-rendering-vs-static-generation (Optional - 10 min)
```
Bit picky but can we use a link label here (e.g. "NextJS Server Side Rendering vs Static Generation") so it looks nice like the other links you've added 😇

adamblanchard commented on react1/week2/lesson-plan-p1.md:
````diff
@@ -0,0 +1,137 @@
+}
+```
+
+## f. Conditional Rendering with Props (5 minutes)
````
"f." typo?

adamblanchard reviewed (changes requested):
Genuinely amazing! Hope it worked well at the weekend too. I guess things will be tweaked over the following weeks but here's a few comments and additional points here:

1. It looks like there are now some unused assets we could remove too? Check each of the assets folders, think there is some in each one
2. Should we have a "homework.md" in each week for consistency? Or remove it from react1 week1 too?
3. Is "react2/readme.md" week 1 topic still accurate? Or should we replace "render props" with "nextjs router"? 🤔

chezzoba commented:
> Genuinely amazing! Hope it worked well at the weekend too. I guess things will be tweaked over the following weeks but here's a few comments and additional points here:
> 
> 1. It looks like there are now some unused assets we could remove too? Check each of the assets folders, think there is some in each one
> 2. Should we have a "homework.md" in each week for consistency? Or remove it from react1 week1 too?
> 3. Is "react2/readme.md" week 1 topic still accurate? Or should we replace "render props" with "nextjs router"? 🤔

Super nice! Good catches and thank you! Yes, this weekend was amazing, thanks to our superstars @bhas and @ddobby94. I addressed these issues.

1. That is very true, I deleted them now. 
2. I removed the one in React 1 Week 1. We now have links in the React 1 readme for that.
3. It is not, and will be addressed - maybe in another branch?

adamblanchard commented:
@chezzoba Fab! I'll approve it for merge, if you can take care of point 3 soon so we don't leave the official docs out of date for too long.

Huge thanks again! Excited to hear how it goes over the next weeks 🙌

P.s. If you don't have permission to merge it, just tell me when and I can do it!
